### ISSUE TITLE
Spike: 

### ISSUE THREAD

**Issue by ckpringle** [vegaprotocol/frontend-monorepo#2881]
## The Chore

Work out what values the user needs to see to feel confident placing a trade 
Assess what we are calculating right now (liquidation price, how is that done, margin remaining...)

## Tasks

- [ ] What do we need to do first
- [ ] and then what?
- [ ] Etc.

**Comment by witgaw** [comment]
I'm not sure liquidation price for an order makes much sense, might be cleaner to just leave it for position.

**Comment by bglownia** [comment]
My findings regarding balance of the margin account.

After order is created balance of margin account is adjusted to required initialMargin. 
InitialMargin can be estimated using estimateOrder query.

To properly estimate initial margin needed after new order will be submitted as size it should be used sum of 
 - size of OrderSubmission
 - openVolume from party positions
 - size from active orders
 - remaining from partially filled orders 
 
Partially filled and active orders pending size do not balance! Margin account balance need to cover initial margin for highest possible volume that can be open if some of orders will be filled. 

E.g. 1
If party has -100 open volume 
If party would like to submit new order with +10 size initial margin is calculated base on -90
If party would like to submit new order with -10 size initial margin is calculated base on -110

E.g. 2
If party has -20 open volume and will open two limit orders for +100 and -20 initial margin is calculated base on +80 size (max sizes are +80 if only first limit order will be filled and -40 if only second limit order will be filled )
If party would like to submit new order with +10 size initial margin is calculated base on +90 size (max sizes are +90 and -40)
If party would like to submit new order with -10 size initial margin is calculated base on +80 size (max sizes are +80 and -50)
If party would like to submit new order with -50 size initial margin is calculated base on -90 size (max sizes are +80 and -90)

**Comment by bglownia** [comment]
If party has already limit orders opened it can happen that it will be not possible to precisely estimate initial margin using  estimateOrder query. This is because orders can have different timeInForce, while timeInForce affect slighly value of initialMargin.

**Comment by bglownia** [comment]
@barnabee @mattrussell36 
Common sense suggest that estimate order query should return total initial margin required after new order will be submitted.
We can then easily compare this with balance on margin account.  
Initial margin without context of active orders and open volume is useless when you already have open volume or active orders. We will need to sent at least two estimate order requests. First to get fees for new order, second to get initial value for sum of open volume, active orders and new order. But we will still not be accurate because we can have active orders with different TIF, and to be accurate we should estimate separately for each TIF. 
Formula how to calculate new initial margin is on the backend already, we just need to use it. We are trying to move calculations to frontend which is error prone and complex. If we leave it on frontend and  backend formula will change we will need to adjust frontend implementation.

**Comment by barnabee** [comment]
> If party has already limit orders opened it can happen that it will be not possible to precisely estimate initial margin using estimateOrder query. This is because orders can have different timeInForce, while timeInForce affect slighly value of initialMargin.

I don't believe this is true. Margin varies between even two identical orders over time and also depending on the size of other orders and positions you have, and also the amount of liquidity/slippage on the book, but they are not impacted by TIF.

There is currently no possibility of a 100% accurate margin estimate, as the calculation is dynamic based on other state that changes in real time (even between requesting the estimate and placing the order).
